DataFormats/interface/WeightedObject.h

```diff
@@ -13,40 +12,18 @@ namespace flashgg {
     {
 
     public:
-        float weight( string key ) const { return ( hasWeight( key ) ? _weights.at( key ) : -1. ); }
+        float weight( string key ) const;
         float centralWeight() const { return weight( central_key ); }
-        void setWeight( string key, float val ) { _weights[key] = val; }
+        void setWeight( string key, float val ) { _labels.push_back( key ); _weights.push_back( val ); }
```

Comment:
As far as I can tell this does not do what you want. 
Any time you tried to set the value of an existing weight you'd append to the back (this would also break the sorting).